src/Service/Resolvers/Sql Query Structures/SqlExecuteQueryStructure.cs

```diff
@@ -44,22 +44,18 @@ public class SqlExecuteStructure : BaseSqlQueryStructure
                 if (requestParams.TryGetValue(paramKey, out object? requestParamValue))
                 {
                     // Parameterize, then add referencing parameter to ProcedureParameters dictionary
-                    try
+                    string? parametrizedName = null;
+                    if (requestParamValue is not null)
                     {
-                        string parameterizedName = MakeParamWithValue(requestParamValue is null ? null :
-                            GetParamAsProcedureParameterType(requestParamValue.ToString()!, paramKey));
-                        ProcedureParameters.Add(paramKey, $"{parameterizedName}");
+                        Type systemType = GetUnderlyingStoredProcedureDefinition().Parameters[paramKey].SystemType!;
+                        parametrizedName = MakeParamWithValue(GetParamAsSystemType(requestParamValue.ToString()!, paramKey, systemType));
```

Comment:
How do we guarantee that instances where the null forgiving operator is used are actually not null?

Reply:
L48 performs a null check on `requestParamValue`. Previously, that null check was not performed and therefore it had to be done when invoking `MakeParamWithValue`.